metals/src/main/scala/scala/meta/internal/rename/RenameProvider.scala

```diff
@@ -86,7 +86,8 @@ final class RenameProvider(
         txtParams <- if (parentSymbols.isEmpty) List(textParams)
         else parentSymbols.map(toTextParams)
         currentReferences = referenceProvider.references(
-          toReferenceParams(txtParams)
+          toReferenceParams(txtParams),
+          includeSynthetics = _.symbol.desc.name.value == "apply"
```

Comment:
Include the synthetics occurrences of `apply` because we want to rename them, but exclude anything else